astropy/coordinates/representation.py

```diff
@@ -223,6 +224,27 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def __eq__(self, other):
+        """
+        Check for equality with another representation.
+        """
+        try:
+            comparisons = [getattr(self, component) ==
+                           getattr(other, component)
+                           for component in self.components]
+        except:
+            try:
+                newrep = other.represent_as(self.__class__)
+                comparisons = [getattr(self, component) ==
+                               getattr(newrep, component)
+                               for component in self.components]
+            except:
+                return False
+        return functools.reduce(operator.and_, comparisons)
```

Comment:
I wanted it to work both for scalar and array-like representations, which will lead to either a list of bool or arrays of `np.bool_` in `comparisons`. Do you think it would be clearer to use `np.logical_and`?